fdmt/cpuFDMT.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3
+
+import sys
+import numpy as np
+from time import time
+from attr import attrs, attrib, cmp_using
+
+@attrs
+class FDMT:
+    fmin: float = attrib(default=400.1953125, help="Frequency of the lowest edge of the band, in MHz")
+    fmax: float = attrib(default=800.1953125, help="Frequency of the highest edge of the band, in MHz")
+    nchan: int = attrib(default=1024, help="Number of frequency channels")
+    maxDT: int = attrib(default=2048, help="Number of time samples corresponding to the maximum delay between the top and bottom of the band (defines the maximum DM of the search)")
+    A: np.ndarray = attrib(default=None, eq=cmp_using(eq=np.array_equal), help="A and B hold the states of successive iterations")
+    B: np.ndarray = attrib(default=None, eq=cmp_using(eq=np.array_equal), help="A and B hold the states of successive iterations")
+    df: float = attrib(init=False)
+    fs: np.ndarray = attrib(init=False, eq=cmp_using(eq=np.array_equal))
+    Q: list = attrib(init=False, eq=cmp_using(eq=np.array_equal), help="Q organizes memory locations of subtransforms for each iteration")
```

Comment:
Maybe this is a silly question, but which version of `attrs` is needed here? I am trying to run the code but Python is not recognising the `help` attribute. (I have attrs v21.4.0.)

Reply:
Correct me if I'm wrong, but I don't think `help` is ever an option to key in in `attrs`

Reply:
Just checked the docs -- looks like you're right

Reply:
oops! Ha, sorry, I've been doing too much `argparse`ing lately!